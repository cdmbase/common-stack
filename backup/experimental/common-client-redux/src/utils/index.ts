export * from './action-helpers';
export * from './context-middleware';
