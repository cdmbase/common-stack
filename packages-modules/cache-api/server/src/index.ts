import module from './module';
import { Feature } from '@common-stack/server-core';

export * from './constants';
export { Cache } from './services';
export default new Feature(module);

