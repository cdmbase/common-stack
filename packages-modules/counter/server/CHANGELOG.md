# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.252-alpha.0](https://github.com/cdmbase/common-stack/compare/v0.0.251...v0.0.252-alpha.0) (2021-03-29)


### Reverts

* Revert "dep break" ([e09ba1e](https://github.com/cdmbase/common-stack/commit/e09ba1e1738f091c7071c807abeeb5f30978a271))
