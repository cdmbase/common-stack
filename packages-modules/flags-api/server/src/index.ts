export * from './constants';
export * from './interfaces';

import * as _ from 'lodash';
import { interfaces } from 'inversify';
import { Feature } from '@common-stack/server-core';

import { ETypes } from './constants';
import { moduleFunc } from './container';
import { storeModuleFunc } from './store';

const createServiceFunc = (container: interfaces.Container) => ({
    flags: container.get(ETypes.FlagsService),
});

export default new Feature({
    schema: [],
    createServiceFunc,
    createContainerFunc: [moduleFunc, storeModuleFunc],
});
