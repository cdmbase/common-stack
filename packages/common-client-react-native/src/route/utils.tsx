import * as React from 'react';
import { IRoute as IRouteProps, IRouteComponentProps } from '@common-stack/client-react';
import { IScreen,  } from '../interfaces';
import { nanoid } from 'nanoid/non-secure';
import { Redirect } from 'react-router-config';

export function flattenRoutes(routes?: IRouteProps[], parent?: IScreen): IScreen[] {
    if (!Array.isArray(routes)) return [];
    const screens: IScreen[] = [];
    for (let idx = 0; idx < routes.length; idx++) {
        const route = routes[idx];
        const { key: routeKey, path, exact, component, strict, redirect, routes: children, ...options } = route;
        const screenKey = routeKey || nanoid();
        const screen: IScreen = {
            key: screenKey,
            name: path || '/',
            component: function ScreenComponent(props) {
                if (redirect) {
                    return <Redirect from={path} to={redirect} exact={exact} strict={strict} />;
                }
                const children = component ? React.createElement(component, props) : null;
                return parent && parent.component ? React.createElement(parent.component, props, children) : children;
            },
            options: {
                ...options,
                routeMatchOpts: route,
            },
        };
        if (Array.isArray(children) && children.length > 0) {
            screens.push(...flattenRoutes(children, screen));
        } else {
            screens.push(screen);
        }
    }
    return screens;
}

