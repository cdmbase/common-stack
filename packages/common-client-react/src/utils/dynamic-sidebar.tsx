import * as React from 'react';
import { Query, withApollo } from 'react-apollo';
import gql from 'graphql-tag';
import Course from '../connector/course';

export function renderQuery(loading, error, data) {
  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;
  return data.allPageHistories.map(allHistory => (
    <Course key={allHistory.id} history={allHistory} />
  ));
}

function AddHistory(props) {
  const { mutate } = props.client;
  const { push, pageNumber } = props;
  mutate({ mutation: gql`
    mutation {
      createPageHistory(number: ${pageNumber}) {
        number
      }
    },
    `, refetchQueries: [{ query: gql`
          {
            allPageHistories(orderBy: id_DESC, first: 5) {
              id
              number
            }
          }
        ` }] }).catch(() => {});
  push(`/history/${pageNumber}`);
  return null;
};

export class History extends React.Component {
  render() {
    return (
      <Query
        query={gql`
          {
            allPageHistories(orderBy: id_DESC, first: 5) {
              id
              number
            }
          }
        `}
      >
        {({ loading, error, data }) => renderQuery(loading, error, data)}
      </Query>
    );
  }
}

export default withApollo(AddHistory as any);

