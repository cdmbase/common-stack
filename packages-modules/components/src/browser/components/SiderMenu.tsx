import * as React from 'react';
import { Link } from 'react-router-dom';
import * as H from 'history';
import { urlToList } from '../utils';
import { Layout, Menu, Icon } from 'antd';
import * as pathToRegexp from 'path-to-regexp';
const { Sider } = Layout;
const { SubMenu } = Menu;

// Allow menu.js config icon as string or ReactNode
//   icon: 'setting',
//   icon: 'http://demo.com/icon.png',
//   icon: <Icon type="setting" />,
const getIcon = (icon, styles = '') => {
    if (typeof icon === 'string' && icon.indexOf('http') === 0) {
        return < img src={icon} alt="icon" className={styles} />;
    } if (typeof icon === 'string') {
        return <Icon type={icon} />;
    }
    return icon;
};

/**
 * Recursively flatten  the data
 * [{path: string}, {path: string}] => {path, path2}
 * @param menu
 */
export const getFlatMenuKeys = menu =>
    menu.reduce((keys, item) => {
        keys.push(item.path);
        if (item.children) {
            return keys.concat(getFlatMenuKeys(item.children));
        }
        return keys;
    }, []);


/**
 * Find all matched menu keys based on paths
 * @param flatMenuKeys: [/abc, /abc/:id, /abc/:id/info]
 * @param paths: [/abc/ /abc/11, /abc/11/info]
 */
export const getMenuMatchKeys = (flatMenuKeys, paths) =>
    paths.reduce((matchKeys, path) => (
        matchKeys.concat(
            flatMenuKeys.filter(item => pathToRegexp(item).test(path)),
        )), []);

export namespace ISiderMenu {
    export interface CompProps {
        menuData: any;
        onCollapse?: any;
        isMobile?: boolean;
        Authorized?: any;
        collapsed?: boolean;
        logo?: any;
        styles?: {
            logo?: any;
            sider?: any;
            icon?: any;
        };
    }

    export interface StateProps {
        location: H.Location;
    }

    export interface CompState {
        openKeys?: any;
    }

    export type Props = CompProps & StateProps;
    export type State = CompState;
}
export default class SiderMenu extends React.PureComponent<ISiderMenu.Props, ISiderMenu.State> {

    private flatMenuKeys;
    private menus;

    constructor(props) {
        super(props);
        this.menus = props.menuData;
        this.flatMenuKeys = getFlatMenuKeys(props.menuData);
        this.state = {
            openKeys: this.getDefaultCollapsedSubMenus(props),
        };
    }
    public componentWillReceiveProps(nextProps) {
        if (nextProps.location.pathname !== this.props.location.pathname) {
            this.setState({
                openKeys: this.getDefaultCollapsedSubMenus(nextProps),
            });
        }
    }

    /**
     * Convert pathname to openKeys
     * /list/search/articles => ['list', '/list/search']
     * @param props
     */
    public getDefaultCollapsedSubMenus(props) {
        const { location: { pathname } } = props || this.props;
        return getMenuMatchKeys(this.flatMenuKeys, urlToList(pathname));
    }

    /**
     * Judge whether it is http link.return or a Link
     * @memberOf SiderMenu
     */
    public getMenuItemPath = item => {
        const { styles } = this.props;
        const itemPath = this.conversionPath(item.path);
        const icon = getIcon(item.icon, styles.icon);
        const { target, name } = item;
        // Is it a http link
        if (/^https?:\/\//.test(itemPath)) {
            return (
                <a href={itemPath} target={target}>
                    {icon}
                    <span>{name}</span>
                </a>
            );
        }
        return (
            <Link
                to={itemPath}
                target={target}
                replace={itemPath === this.props.location.pathname}
                onClick={
                    this.props.isMobile
                        ? () => {
                            this.props.onCollapse(true);
                        }
                        : undefined
                }
            >
                {icon}
                <span>{name}</span>
            </Link>
        );
    }
    /**
     * get SubMenu or Item
     */
    public getSubMenuOrItem = item => {
        const { styles } = this.props;
        if (item.children && item.children.some(child => child.name)) {
            const childrenItems = this.getNavMenuItems(item.children);
            if (childrenItems && childrenItems.length > 0) {
                return (
                    <SubMenu
                        title={
                            item.icon ? (
                                <span>
                                    {getIcon(item.icon, styles.icon)}
                                    <span>{item.name}</span>
                                </span>
                            ) : (
                                    item.name
                                )
                        }
                        key={item.path}
                    >
                        {childrenItems}
                    </SubMenu>
                );
            }
            return null;
        } else {
            return <Menu.Item key={item.path}>{this.getMenuItemPath(item)}</Menu.Item>;
        }
    }
    /**
     * @memberof SiderMenu
     */
    public getNavMenuItems = menusData => {
        if (!menusData) {
            return [];
        }
        return menusData.filter(item => item.name && !item.hideInMenu)
            .map(item => {
                // make dom
                const ItemDom = this.getSubMenuOrItem(item);
                return this.checkPermissionItem(item.authority, ItemDom);
            })
            .filter(item => item);
    }

    // Get the currently selected menu
    public getSelectedMenuKeys = () => {
        const { location: { pathname } } = this.props;
        return getMenuMatchKeys(this.flatMenuKeys, urlToList(pathname));
    }
    // conversion Path
    public conversionPath = path => {
        if (path && path.indexOf('http') === 0) {
            return path;
        } else {
            return `/${path || ''}`.replace(/\/+/g, '/');
        }
    }
    // permission to check
    public checkPermissionItem = (authority, ItemDom) => {
        if (this.props.Authorized && this.props.Authorized.check) {
            const { check } = this.props.Authorized;
            return check(authority, ItemDom);
        }
        return ItemDom;
    }
    public isMainMenu = key => {
        return this.menus.some(item => key && (item.key === key || item.path === key));
    }
    public handleOpenChange = openKeys => {
        const lastOpenKey = openKeys[openKeys.length - 1];
        const moreThanOne = openKeys.filter(openKey => this.isMainMenu(openKey)).length > 1;
        this.setState({
            openKeys: moreThanOne ? [lastOpenKey] : [...openKeys],
        });
    }

    public render() {
        const { logo, collapsed, onCollapse, styles } = this.props;
        const { openKeys } = this.state;
        // Don't show popup menu when it is been collapsed
        const menuProps = collapsed ? {} : { openKeys };
        // If pathname can't match, use the nearest parent's key
        let selectedKeys = this.getSelectedMenuKeys();
        if (!selectedKeys.length) {
            selectedKeys = [openKeys[openKeys.length - 1]];
        }

        return (
            <Sider
                trigger={null}
                collapsible={true}
                collapsed={collapsed}
                breakpoint="lg"
                onCollapse={onCollapse}
                width={256}
                className={styles.sider}
            >
                <div className={styles.logo} key="logo">
                    <Link to="/">
                        <img src={logo} alt="logo" />
                        <h1>Ant Design Pro</h1>
                    </Link>
                </div>
                <Menu
                    key="Menu"
                    theme="dark"
                    mode="inline"
                    {...menuProps}
                    onOpenChange={this.handleOpenChange}
                    selectedKeys={selectedKeys}
                    style={{ padding: '16px 0', width: '100%' }}
                >
                    {this.getNavMenuItems(this.menus)}
                </Menu>
            </Sider>
        );
    }

}

