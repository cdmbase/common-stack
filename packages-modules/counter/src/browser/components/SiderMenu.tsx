import * as React from 'react';
import SiderMenu from '@common-stack/components/lib/browser/components/SiderMenu';


const genMenu = {
    ['/dashboard']: { name: 'dashboard' },
    ['/dashboard/name']: { name: 'name' },
    ['/userinfo']: { name: 'userinfo' },
    ['/userinfo/:id']: { name: 'userinfo-id' },
    ['/userinfo/:id/info']: { name: 'user-info' },
};



export default ({ menuData, location }) => (
    <div>
        <SiderMenu
            menuData={menuData}
            location={location}
            isMobile={false}
            styles={{ sider: '' }}
        />
    </div>
);


