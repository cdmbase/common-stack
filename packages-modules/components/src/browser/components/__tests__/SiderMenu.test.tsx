import * as React from 'react';
import { urlToList } from '../../utils';
import SiderMenu, { getFlatMenuKeys, getMenuMatchKeys, ISiderMenu } from '../SiderMenu';
import { Feature } from '@common-stack/client-react';
import * as ReactDOMServer from 'react-dom/server';
import StaticRouter from 'react-router/StaticRouter';
import 'jest';

const genMenu = {
    ['/dashboard']: {},
    ['/dashboard/name']: {},
    ['/userinfo']: {},
    ['/userinfo/:id']: {},
    ['/userinfo/:id/info']: {},
};


const connector = new Feature({ menuConfig: genMenu });
const menu = connector.getMenus();

const flatMenuKeys = getFlatMenuKeys(menu);


describe('test convert nested menu to flat menu', () => {
    it('simple menu', () => {

        expect(flatMenuKeys).toEqual([
            '/dashboard',
            '/dashboard/name',
            '/userinfo',
            '/userinfo/:id',
            '/userinfo/:id/info',
        ]);
    });
});

describe('test menu match', () => {
    it('simple path', () => {
        expect(getMenuMatchKeys(flatMenuKeys, urlToList('/dashboard'))).toEqual(['/dashboard']);
    });

    it('error path', () => {
        expect(getMenuMatchKeys(flatMenuKeys, urlToList('/dashboardname'))).toEqual([]);
    });

    it('Secondary path', () => {
        expect(getMenuMatchKeys(flatMenuKeys, urlToList('/dashboard/name'))).toEqual([
            '/dashboard',
            '/dashboard/name',
        ]);
    });

    it('Parameter path', () => {
        expect(getMenuMatchKeys(flatMenuKeys, urlToList('/userinfo/2144'))).toEqual([
            '/userinfo',
            '/userinfo/:id',
        ]);
    });

    it('three parameter path', () => {
        expect(getMenuMatchKeys(flatMenuKeys, urlToList('/userinfo/2144/info'))).toEqual([
            '/userinfo',
            '/userinfo/:id',
            '/userinfo/:id/info',
        ]);
    });


});

describe('test component', () => {


    it('sidermenu', () => {

        const pathname = '/dashboard';
        const component = ReactDOMServer.renderToString(
            <StaticRouter location={pathname} context={{}}>
                <SiderMenu
                    menuData={menu}
                    isMobile={false}
                    location={{ pathname } as any}
                    styles={{ sider: '' }}
                />
            </StaticRouter>,
        );

        expect(component).toMatchSnapshot();

    });
});

