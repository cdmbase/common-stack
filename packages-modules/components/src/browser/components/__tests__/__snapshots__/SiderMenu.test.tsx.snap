// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component sidermenu 1`] = `"<div class=\\"ant-layout-sider ant-layout-sider-dark\\" style=\\"flex:0 0 256px;max-width:256px;min-width:256px;width:256px\\" data-reactroot=\\"\\"><div class=\\"ant-layout-sider-children\\"><div><a href=\\"/\\"><img alt=\\"logo\\"/><h1>Ant Design Pro</h1></a></div><ul style=\\"padding:16px 0;width:100%\\" class=\\"ant-menu ant-menu-dark ant-menu-root ant-menu-inline\\" role=\\"menu\\"></ul></div></div>"`;
